Source/WebKit/UIProcess/API/C/WKPage.cpp

```diff
@@ -2496,7 +2496,7 @@ void WKPageSetPageNavigationClient(WKPageRef pageRef, const WKPageNavigationClie
         void legacyWebCryptoMasterKey(WebKit::WebPageProxy& page, CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&& completionHandler) override
         {
             if (m_client.copyWebCryptoMasterKey) {
-                if (auto data = adoptRef(toImpl(m_client.copyWebCryptoMasterKey(toAPI(&page), m_client.base.clientInfo))))
+                if (auto data = toProtectedImpl(m_client.copyWebCryptoMasterKey(toAPI(&page), m_client.base.clientInfo)))
```

Comment:
This looks like either the previous code was wrong or you are now leaking memory since we no longer adopting. I suspect you are leaking memory given that the function we're calling is named "copy*".